homeassistant/components/swidget/coordinator.py

```diff
@@ -0,0 +1,54 @@
+"""Swidget Device Coordinator."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from swidget import SwidgetDevice, SwidgetException
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+REQUEST_REFRESH_DELAY = 1.0
+
+
+class SwidgetDataUpdateCoordinator(DataUpdateCoordinator):
+    """DataUpdateCoordinator to gather data for a specific Swidget device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        device: SwidgetDevice,
+    ) -> None:
+        """Initialize DataUpdateCoordinator to gather data for specific device."""
+        self.device = device
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=device.ip_address,
+            always_update=False,
+            update_interval=timedelta(seconds=300),
+            request_refresh_debouncer=Debouncer(
+                hass, _LOGGER, cooldown=0.35, immediate=False
+            ),
+        )
+
+    async def async_initialize(self) -> Any:
+        """Initialize a callback for any websocket events received from the device."""
+        return self.device.add_event_callback(self.websocket_event_callback)
```

Comment:
Iit's a bit weird that you're doing some initialization inside the init, but then some here. I'd move it in one place